x-pack/plugins/profiling/public/components/flamegraph_nav.tsx

```diff
@@ -6,17 +6,30 @@
  */
 
 import React, { useEffect } from 'react';
+import { ElasticFlameGraph } from '../../common/flamegraph';
+import { TimeRange } from '../../common/types';
 
-export const FlameGraphNavigation = ({ index, projectID, n, timeRange, getter, setter }) => {
+interface Props {
+  index: string;
+  projectID: number;
+  n: number;
+  timeRange: TimeRange;
+  getter: (
+    index: string,
+    projectID: number,
+    unixStart: number,
+    unixEnd: number,
+    n: number
+  ) => Promise<ElasticFlameGraph>;
+  setter: React.Dispatch<ElasticFlameGraph>;
+}
+
+export const FlameGraphNavigation = ({ index, projectID, n, timeRange, getter, setter }: Props) => {
   useEffect(() => {
-    console.log(new Date().toISOString(), 'started payload retrieval');
     getter(index, projectID, timeRange.unixStart, timeRange.unixEnd, n).then((response) => {
-      console.log(new Date().toISOString(), 'finished payload retrieval');
-      console.log(new Date().toISOString(), response);
       setter(response);
-      console.log(new Date().toISOString(), 'updated local state');
     });
-  }, [index, projectID, n, timeRange]);
+  }, [index, projectID, n, timeRange, getter, setter]);
```

Comment:
The getter is from a service, and the setter is from a hook managed by React, so its identity should remain the same over renders. Adding these shouldn't change the behaviour of the `useEffect` hook but it makes the linter happy.